core/src/main/java/org/apache/druid/java/util/common/StringUtils.java

```diff
@@ -481,9 +482,10 @@ public static String repeat(String s, int count)
    *
    * @return the string left-padded with pad to a length of len
    */
-  public static String lpad(String base, Integer len, String pad)
+  @Nullable
+  public static String lpad(@Nonnull String base, int len, @Nonnull String pad)
   {
-    if (len < 0) {
+    if (len < 0 || pad.isEmpty()) {
```

Comment:
Doesn't `rpad()` need a similar change?

Reply:
Yeah I thought I made the change because FunctionTest was passing. Fixed now.

I debugged it, and saw that FunctionTest passes because it treats the empty string as null ( probably the default null handling mode in the Expr system), so the rpad function thinks the `pad` that is passed in is null, not empty.